tree/forest/v7/src/RField.cxx

```diff
@@ -0,0 +1,543 @@
+   // TODO(jblomer): use a type translation map
+   if (normalizedType == "Float_t") normalizedType = "float";
+   if (normalizedType == "Double_t") normalizedType = "double";
+   if (normalizedType == "Int_t") normalizedType = "std::int32_t";
+   if (normalizedType == "int") normalizedType = "std::int32_t";
+   if (normalizedType == "unsigned") normalizedType = "std::uint32_t";
+   if (normalizedType == "unsigned int") normalizedType = "std::uint32_t";
+   if (normalizedType == "UInt_t") normalizedType = "std::uint32_t";
+   if (normalizedType == "ULong64_t") normalizedType = "std::uint64_t";
+   if (normalizedType == "string") normalizedType = "std::string";
+   if (normalizedType.substr(0, 7) == "vector<") normalizedType = "std::" + normalizedType;
+
+   if (normalizedType == "ROOT::Experimental::ClusterSize_t") return new RField<ClusterSize_t>(fieldName);
+   if (normalizedType == "std::int32_t") return new RField<std::int32_t>(fieldName);
+   if (normalizedType == "std::uint32_t") return new RField<std::uint32_t>(fieldName);
+   if (normalizedType == "std::uint64_t") return new RField<std::uint64_t>(fieldName);
+   if (normalizedType == "float") return new RField<float>(fieldName);
+   if (normalizedType == "double") return new RField<double>(fieldName);
+   if (normalizedType == "std::string") return new RField<std::string>(fieldName);
+   if (normalizedType.substr(0, 12) == "std::vector<") {
+      std::string itemTypeName = normalizedType.substr(12, normalizedType.length() - 13);
+      auto itemField = Create(GetCollectionName(fieldName), itemTypeName);
+      return new RFieldVector(fieldName, std::unique_ptr<Detail::RFieldBase>(itemField));
+   }
+   // For the time being, we silently read RVec fields as std::vector
+   if (normalizedType.substr(0, 19) == "ROOT::VecOps::RVec<") {
+      std::string itemTypeName = normalizedType.substr(19, normalizedType.length() - 20);
+      auto itemField = Create(GetCollectionName(fieldName), itemTypeName);
+      return new RFieldVector(fieldName, std::unique_ptr<Detail::RFieldBase>(itemField));
+   }
+   // TODO: create an RFieldCollection?
+   if (normalizedType == ":Collection:") return new RField<ClusterSize_t>(fieldName);
+   auto cl = TClass::GetClass(normalizedType.c_str());
+   if (cl != nullptr) {
+      return new RFieldClass(fieldName, normalizedType);
+   }
+   R__ASSERT(false);
+   return nullptr;
+}
+
+void ROOT::Experimental::Detail::RFieldBase::DoAppend(const ROOT::Experimental::Detail::RFieldValueBase& /*value*/) {
+   R__ASSERT(false);
+}
+
+void ROOT::Experimental::Detail::RFieldBase::DoRead(
+   ROOT::Experimental::ForestSize_t /*index*/,
+   RFieldValueBase* /*value*/)
+{
+   R__ASSERT(false);
+}
+
+void ROOT::Experimental::Detail::RFieldBase::DoReadV(
+   ROOT::Experimental::ForestSize_t /*index*/,
+   ROOT::Experimental::ForestSize_t /*count*/,
+   void* /*dst*/)
+{
+   R__ASSERT(false);
```

Comment:
Why implement the function if it is never to be called (shouldn't it be an abstract function?)